src/precice/impl/ParticipantImpl.hpp

```diff
@@ -363,12 +376,14 @@ class ParticipantImpl {
   /// Helper for mapWrittenData and mapReadData
   void computeMappings(std::vector<MappingContext> &contexts, const std::string &mappingType);
 
-  /// Computes, performs, and resets all suitable write mappings.
-  void mapWrittenData();
+  /// Computes, and performs suitable write mappings either entirely or after given time
+  void mapWrittenData(std::optional<double> after = std::nullopt);
 
-  /// Computes, performs, and resets all suitable read mappings.
+  // Computes, and performs read mappings optionally removes samples in toData after the optional time
```

Comment:
```suggestion
  // Computes and performs read mappings. Optionally removes samples in toData after the optional time.
```

Is the second part of the comment true? Which optional time do you mean?